x-pack/legacy/plugins/apm/public/components/app/Main/route_config/index.tsx

```diff
@@ -107,6 +110,74 @@ export const routes: BreadcrumbRoute[] = [
     ),
     name: RouteName.AGENT_CONFIGURATION
   },
+
+  {
+    exact: true,
+    path: '/settings/agent-configuration/create',
+    breadcrumb: i18n.translate(
+      'xpack.apm.breadcrumb.settings.createAgentConfigurationTitle',
+      { defaultMessage: 'Create Agent Configuration' }
+    ),
+    name: RouteName.AGENT_CONFIGURATION_CREATE,
+    component: () => {
+      const { search } = history.location;
+
+      // Ignoring here because we specifically DO NOT want to add the query params to the global route handler
+      // @ts-ignore
+      const { pageStep } = toQuery(search);
```

Comment:
Controversy `#2`: not adding query params to the global url params. 
I want to get us away from having global url params (query params) and think of query and path params per route (just like on the server side. This means the route should be responsible for specifying the params it needs.

Reply:
I've updated https://github.com/elastic/kibana/issues/51963 with my suggestions for improvements to route params, linking to routes and retrieving route params from within components